Wabbajack/AppState.cs

```diff
@@ -262,69 +381,198 @@ private void UpdateLoop()
+                        while (!_cachedSlides.ContainsKey(randomElement.ModID) || slidesQueue.Contains(randomElement))
                         {
-                            var element = SlideShowElements[idx];
+                            idx = _random.Next(0, SlideShowElements.Count);
+                            randomElement = SlideShowElements[idx];
+                        }
 
-                            var data = new MemoryStream();
-                            using (var stream = new HttpClient().GetStreamSync("asdas" + element.ImageURL))
-                                stream.CopyTo(data);
-                            data.Seek(0, SeekOrigin.Begin);
+                        if (_cachedSlides.ContainsKey(randomElement.ModID))
+                        {
+                            var bitmap = new BitmapImage();
+                            _cachedSlides.TryGetValue(randomElement.ModID, out bitmap);
+                            bitmap = null;
+                            _cachedSlides.Remove(randomElement.ModID);
+                            //hmmm
+                            //GC.Collect();
+                        }
+                    } while (_cachedSlides.Count >= MAX_CACHE_SIZE);
+                }
 
+                SlideShowItem element = slidesQueue.Peek();
+                if (!element.Adult || (element.Adult && SplashShowNSFW))
+                {
+                    SplashScreenImage = _noneImage;
+                    if (element.ImageURL != null)
+                    {
 
-                            dispatcher.Invoke(() =>
+                        dispatcher.Invoke(() =>
+                        {
+                            if (_cachedSlides.ContainsKey(element.ModID))
                             {
                                 var bitmap = new BitmapImage();
-                                bitmap.BeginInit();
-                                bitmap.CacheOption = BitmapCacheOption.OnLoad;
-                                bitmap.StreamSource = data;
-                                bitmap.EndInit();
-
+                                _cachedSlides.TryGetValue(element.ModID, out bitmap);
                                 SplashScreenImage = bitmap;
-                                SplashScreenModName = element.ModName;
-                                SplashScreenAuthorName = element.AuthorName;
-                                SplashScreenSummary = element.ModSummary;
-                                _nexusSiteURL = element.ModURL;
+                            }
+                        });
+                    }
+                    _originalImage = false;
+                    SplashScreenModName = element.ModName;
+                    SplashScreenAuthorName = element.AuthorName;
+                    SplashScreenSummary = element.ModSummary;
+                    _nexusSiteURL = element.ModURL;                  
+                }
+                if (fromLoop)
```

Comment:
Hmm, do we want to check this here?   By only updating the timestamp when it's an update triggered by the loop, you may get a situation where the user manually hits "next", which triggers a flip, but then the loop timer fires right after, leading to a second spammy flip.

I'd think we'd want to reset the timestamp regardless of where the update came from.  Any update, whether auto or manual click should make the system wait the X seconds before doing another auto-flip.